### PR TITLE
cyclonedx-python: 3.11.7 -> 4.5.0

### PR DESCRIPTION
**Pull request by xanderio**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by xanderio** [comment]
Result of `nixpkgs-review pr 329874` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>cyclonedx-python</li>
    <li>cyclonedx-python.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>python311Packages.cyclonedx-python-lib</li>
    <li>python311Packages.cyclonedx-python-lib.dist</li>
    <li>python312Packages.cyclonedx-python-lib</li>
    <li>python312Packages.cyclonedx-python-lib.dist</li>
  </ul>
</details>

**Comment by kirillrdy** [comment]
@xanderio correct commit message should be `cyclonedx-python: 3.11.7 -> 4.5.0` or `python312Packages.cyclonedx-python-lib: 3.11.7 -> 4.5.0`   otherwise ofborg will not build anything

**Comment by xanderio** [comment]
@kirillrdy fixed

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 1]
```diff
@@ -1,65 +1,43 @@
-{ lib
-, fetchFromGitHub
-, python3
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+  nix-update-script,
```
Comment (fabaff) [inline pkgs/tools/misc/cyclonedx-python/default.nix:5]: I don't think that we need the update-script as the update process is straight-forward and `nix-update` will work.

File: pkgs/development/python-modules/cyclonedx-python-lib/default.nix [hunk 2]
```diff
@@ -49,6 +49,7 @@ buildPythonPackage rec {
     py-serializable
     types-setuptools
     types-toml
+    jsonschema
```
Comment (fabaff) [inline pkgs/development/python-modules/cyclonedx-python-lib/default.nix:52]: We should start using `optional-dependencies`and follow the upstream requirements (https://github.com/CycloneDX/cyclonedx-python-lib/blob/main/pyproject.toml#L78-L81).

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 3]
```diff
@@ -1,65 +1,43 @@
...
     repo = "cyclonedx-python";
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-jU/0FkQCyph59TnEE+lckJXsU9whfvWp7dkdfzprYw8=";
+    hash = "sha256-+XeMRREDX1+v+qOeYiHh7uhadfueYYOxspLY3q1NL6s=";
   };
 
-  nativeBuildInputs = with py.pkgs; [
-    poetry-core
-  ];
+  nativeBuildInputs = with python3Packages; [ poetry-core ];
```
Comment (fabaff) [inline pkgs/tools/misc/cyclonedx-python/default.nix:20]:
```suggestion
  build-system = with python3Packages; [ poetry-core ];
```

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 4]
```diff
@@ -1,65 +1,43 @@
...
 
-  propagatedBuildInputs = with py.pkgs; [
+  propagatedBuildInputs = with python3Packages; [
```
Comment (fabaff) [inline pkgs/tools/misc/cyclonedx-python/default.nix:22]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 5]
```diff
@@ -1,65 +1,43 @@
...
     chardet
     cyclonedx-python-lib
     packageurl-python
     pip-requirements-parser
-    setuptools
-    toml
+    packaging
+    tomli
   ];
```
Comment (fabaff) [inline pkgs/tools/misc/cyclonedx-python/default.nix:29]: See above

```suggestion
  ] ++ cyclonedx-python-lib.optional-dependencies.validation;
```

https://github.com/CycloneDX/cyclonedx-python/blob/main/pyproject.toml#L72C1-L72C80

**Comment by xanderio** [comment]
@ofborg build python312Packages.cyclonedx-python-lib

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 329874` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>cyclonedx-python</li>
    <li>cyclonedx-python.dist</li>
    <li>nixpkgs-manual</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Change looks good to me.

**Pull request merged**